src/test/java/io/localmotion/batch/notification/ChatBoxNotificationBatchTest.java

```diff
@@ -0,0 +1,116 @@
+package io.localmotion.batch.notification;
+
+import io.localmotion.chatbox.ChatMessage;
+import io.localmotion.chatbox.ChatboxRepository;
+import io.localmotion.chatbox.IChatboxRepository;
+import io.localmotion.chatbox.notification.ChatBoxNotification;
+import io.localmotion.chatbox.notification.ChatBoxNotificationId;
+import io.localmotion.chatbox.notification.ChatBoxNotificationRepository;
+import io.localmotion.chatbox.notification.IChatBoxNotificationRepository;
+import io.localmotion.personaldata.IPersonalDataRepository;
+import io.localmotion.personaldata.PersonalDataRecord;
+import io.localmotion.personaldata.PersonalDataRepository;
+import io.micronaut.context.annotation.Primary;
+import io.micronaut.test.annotation.MicronautTest;
+import io.micronaut.test.annotation.MockBean;
+import org.junit.jupiter.api.Test;
+
+import javax.inject.Inject;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@MicronautTest
+class ChatBoxNotificationBatchTest {
+
+	@Inject
+	ChatBoxNotificationBatch chatBoxNotificationBatch;
+
+	@Inject
+	IChatBoxNotificationRepository chatBoxNotificationRepository;
+
+	@Inject
+	IChatboxRepository chatboxRepository;
+
+	@Inject
+	IPersonalDataRepository personalDataRepository;
+
+	@Test
+	void notifyUnreadMessages() throws InterruptedException {
```

Comment:
What exactly are you testing here? Aren't you just testing the mock services?

Reply:
Basically class `ChatBoxNotificationBatchTest` is under the test by mocking its dependents. 

I will write the Integration test that will test end to end flow without mocking its dependents.